README.md

```diff
@@ -64,3 +64,90 @@ consumer = Consumer(client_id='fill me in',
+consumer.subscribe(topics)
+
+while True:
+    for message in consumer.consume():
+        print(message.value())
+        consumer.commit(message)
+```
+
+**How can I read messages beginning at the earliest available messages for a given stream?**
+
+You can begin reading a given topic stream from the earliest message that is present in the stream buffer by setting the Group ID to an empty string and applying the ‘earliest’ setting for the auto offset reset option in the configuration dictionary argument for the Consumer class. This feature allows the user to scan for older messages for testing purposes or to recover messages that may have been missed due to a crash or network outage. Just keep in mind that the stream buffers are finite in size. They currently hold messages from the past few days.
+
+Example code:
+```python3
+from gcn_kafka import Consumer
+
+config = {'group.id': '',
+          'auto.offset.reset': 'earliest'}
+
+consumer = Consumer(config=config,
+                    client_id='fill me in',
+                    client_secret='fill me in',
+                    domain='gcn.nasa.gov')
+
+topics = ['gcn.classic.voevent.INTEGRAL_SPIACS']
+consumer.subscribe(topics)
+
+while True:
+    for message in consumer.consume():
+        print(message.value())
+```
+
+Note: Adding a timeout argument, given as an integer number of seconds, to consume() will allow the program to exit quickly once it has reached the end of the existing message buffer. This is useful for users who just want to recover an older message from the stream.
+
+**How can I search for messages occurring within a given date range?**
+
+To search for messages in a given date range, set the Group ID to an empty string and the auto offset reset option to the ‘earliest’ setting in the configuration dictionary argument for the Consumer class. Then use the timestamp member function to select for messages occurring within a given date range. When doing so, keep in mind that the stream buffers are finite in size. It is not possible to recover messages prior to the start of the stream buffer. The GCN stream buffers are currently set to hold messages from the past few days.
+
+Example code:
+```python3
+import datetime
+from gcn_kafka import Consumer
+
+config = {'group.id': '',
+          'auto.offset.reset': 'earliest'}
+
+consumer = Consumer(config=config,
+                    client_id='fill me in',
+                    client_secret='fill me in',
+                    domain='gcn.nasa.gov')
+
+topics = ['gcn.classic.voevent.INTEGRAL_SPIACS']
+consumer.subscribe(topics)
+
+# search for messages occurring 3 days ago
+date1 = datetime.datetime.now() - datetime.timedelta(days=3)
+date2 = date1 + datetime.timedelta(days=1)
+
+while True:
+    message = consumer.poll()
+    date = datetime.datetime.fromtimestamp(message.timestamp()[1]/1000.)
+    if date1 < date:
+        if date > date2:
+            break
+        print(message.value())
+```
```

Comment:
Hmm, a linear search of the whole topic? Seems inefficient. Is `Consumer.offsets_for_times()` any help here?

Reply:
Ok, I wasn't aware of the offsets_for_times() function. That's a better way than the linear search. I'll test it & update the example.

Reply:
@lpsinger can you take a look at the most recent commit? I changed the last FAQ example to use `Consumer.offsets_for_times()` since it is indeed simpler and more efficient than the linear search to just assign the stream start to the time a user wants and consume the desired number of messages. I had to do two separate calls to `offsets_for_times()` to get the start & end of the desired date range because the list entries were behaving like pointers for arguments that had the same topic name.

Reply:
Did it work?

Reply:
Yup, the updated example successfully pulls in all the messages within the requested date range without needing to do a search via the loop. It just grabs the messages directly by starting at the first offset value. Here is the output from when I added a few extra print statements to check: 
```
start time:  2022-08-06 13:17:06.281000
end time:    2022-08-07 13:17:06.281000
<cimpl.Message object at 0x103df9440>
consumed message with timestamp  2022-08-06 14:00:02.604000
<cimpl.Message object at 0x103df94c0>
consumed message with timestamp  2022-08-06 20:00:04.316000
<cimpl.Message object at 0x103df95c0>
consumed message with timestamp  2022-08-07 02:00:04.178000
<cimpl.Message object at 0x103df9540>
consumed message with timestamp  2022-08-07 08:00:04.541000

Reply:
Cool!